CRM/Core/Form.php

```diff
@@ -1584,7 +1584,7 @@ public static function validateMandatoryFields($fields, $values, &$errors) {
    */
   public function getContactID() {
     $tempID = CRM_Utils_Request::retrieve('cid', 'Positive', $this);
-    if (isset($this->_params) && isset($this->_params['select_contact_id'])) {
+    if (isset($this->_params) && !empty($this->_params['select_contact_id'])) {
```

Comment:
Would this not better be (don't assume select_contact_id is set):
 if (isset($this->_params, $this->_params['select_contact_id']) && !empty($this->_params['select_contact_id'])) {